tests/tests-memcached/src/main/java/com/hazelcast/simulator/memcached/ReadWriteTest.java

```diff
@@ -0,0 +1,132 @@
+import net.spy.memcached.transcoders.SerializingTranscoder;
+
+import java.util.Random;
+
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateByteArray;
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateStrings;
+
+public class ReadWriteTest extends MemcachedTest {
+
+    // properties
+    public int keyCount = 1000;
+    public int keyLength = 10;
+    public int valueCount = 1000;
+    public int minSize = 16;
+    public int maxSize = 2000;
+    public int ttl = 0; // by default insert immortal entries
+    // the number of keys that are going to be written.
+    // normally you want to keep this the same as keyCount (for reading), but it can help to expose certain problems like
+    // gc. If they writeKeyCount is very small, only a small group of objects get updated frequently and helps to prevent
+    // getting them tenured. If writeKeyCount is -1, it will automatically be set to keyCount
+    public int writeKeyCount = -1;
+
+    private String[] keys;
+    private byte[][] values;
+
+    @Setup
+    public void setUp() {
+        keys = generateStrings(keyCount, keyLength);
+
+        if (minSize > maxSize) {
+            throw new IllegalStateException("minSize can't be larger than maxSize");
+        }
+
+        if (writeKeyCount == -1) {
+            writeKeyCount = keyCount;
+        }
+    }
+
+    @Prepare
+    public void prepare() {
+        Random random = new Random();
+        values = new byte[valueCount][];
+        for (int i = 0; i < values.length; i++) {
+            int delta = maxSize - minSize;
+            int length = delta == 0 ? minSize : minSize + random.nextInt(delta);
+            values[i] = generateByteArray(random, length);
+        }
+
+        for (String key : keys) {
+            cluster.add(key, ttl, values[random.nextInt(values.length)]);
+        }
+    }
+
+    @TimeStep(prob = 0.1)
+    public OperationFuture<Boolean> put(ThreadState state) {
+        return cluster.add(state.randomKey(), ttl, state.randomValue(), state.serializingTranscoder());
+    }
+
+    @TimeStep(prob = 0.0)
+    public boolean putWithCheck(ThreadState state) throws Exception {
+        return cluster.add(state.randomKey(), ttl, state.randomValue(), state.serializingTranscoder()).get();
+    }
+
+    @TimeStep(prob = 0.0)
+    public OperationFuture<Boolean> set(ThreadState state) {
+        return cluster.set(state.randomWriteKey(), ttl, state.randomValue(), state.serializingTranscoder());
+    }
+
+    @TimeStep(prob = 0.0)
+    public boolean setWithCheck(ThreadState state) throws Exception {
+        return cluster.set(state.randomWriteKey(), ttl, state.randomValue(), state.serializingTranscoder()).get();
+    }
+
+    @TimeStep(prob = -1)
+    public Object get(ThreadState state) {
+        return cluster.get(state.randomKey());
+    }
+
+    public class ThreadState extends BaseThreadState {
+
+        private SerializingTranscoder st;
```

Comment:
don't use short names. Harder to understand. What I normally do is find a balance e.g. transcoder.